dotCMS/src/integration-test/java/com/dotcms/contenttype/test/ContentTypeAPIImplTest.java

```diff
@@ -787,8 +787,215 @@ public void testWidgetContentTypeWithPublishExpireFields() throws Exception{
+        
+        Field fieldToSave = FieldBuilder.builder( DateTimeField.class )
+                .name( "Publish Date" )
+                .variable( "publishDate" )
+                .contentTypeId( contentType.id() )
+                .dataType( DataTypes.DATE )
+                .indexed(true)
+                .build();
+        fields.add( fieldToSave );
+        
+        fieldToSave = FieldBuilder.builder( DateTimeField.class )
+                .name( "Expire Date" )
+                .variable( "expireDate" )
+                .contentTypeId( contentType.id() )
+                .dataType( DataTypes.DATE )
+                .indexed(true)
+                .build();		
+        fields.add( fieldToSave );
+        
+        contentType = contentTypeApi.save(contentType, fields);
+        
+        //Deleting content type.
+      	delete(contentType);
+	}
+	
+	/*
+	 * Github: https://github.com/dotCMS/core/issues/11861
+	 * 
+	 * Creates a File and a couple of DateTimeFields (Publish and Expire) and set it as Publish and Expire properties in the Content Type.
+	 */
+	@Test
+	public void testFileContentTypeWithPublishExpireFields() throws Exception{
+		int base = BaseContentType.FILEASSET.ordinal();
+		long time = System.currentTimeMillis();
+		
+		ContentType contentType = ContentTypeBuilder
+                .builder(BaseContentType.getContentTypeClass(base))
+                .description("FileContentTypeWithPublishExpireFields " + time)
+                .folder(FolderAPI.SYSTEM_FOLDER)
+                .host(Host.SYSTEM_HOST)
+                .name("FileContentTypeWithPublishExpireFields " + time)
+                .owner(APILocator.systemUser().toString())
+                .variable("FileCTVariable")
+                .publishDateVar("publishDate")
+                .expireDateVar("expireDate")
+                .build();
+        contentType = contentTypeApi.save(contentType);
+
+        assertThat("ContentType exists", contentTypeApi.find( contentType.inode() ) != null);
+        
+        List<Field> fields = new ArrayList<>( contentType.fields() );
+        
+        Field fieldToSave = FieldBuilder.builder( DateTimeField.class )
+                .name( "Publish Date" )
+                .variable( "publishDate" )
+                .contentTypeId( contentType.id() )
+                .dataType( DataTypes.DATE )
+                .indexed(true)
+                .build();
+        fields.add( fieldToSave );
+        
+        fieldToSave = FieldBuilder.builder( DateTimeField.class )
+                .name( "Expire Date" )
+                .variable( "expireDate" )
+                .contentTypeId( contentType.id() )
+                .dataType( DataTypes.DATE )
+                .indexed(true)
+                .build();		
+        fields.add( fieldToSave );
+        
+        contentType = contentTypeApi.save(contentType, fields);
+        
+      //Deleting content type.
+      	delete(contentType);
```

Comment:
Almost the entire code of the test is repeated in the following tests. This can be extracted to a private method and pass through parameters the things that change between tests (I think only passing the BaseType will suffice)